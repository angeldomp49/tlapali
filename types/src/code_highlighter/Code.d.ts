import { Lang, Theme } from "@makechtec/tlapali_dev";
export declare const Code: (props: {
    theme: Theme;
    lang: Lang;
    children: any;
}) => JSX.Element;
