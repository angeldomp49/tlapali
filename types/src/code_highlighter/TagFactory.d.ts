export declare class TagFactory {
    createStartTag(tagName: string, extraAttributes?: string): string;
    createEndTag(tagName: string): string;
    createStyleAttribute(style: string): string;
    encloseTextWithTag(text: string, tagName: string, extraAttributes?: string): string;
}
