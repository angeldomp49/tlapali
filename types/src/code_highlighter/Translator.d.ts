import { Lang, Theme } from "@makechtec/tlapali_dev";
import { TagFactory } from "./TagFactory";
export declare class Translator {
    private theme;
    private lang;
    private tagFactory;
    private tagName;
    private supportedLang;
    constructor(theme: Theme, lang: Lang);
    convert(code: string): string;
    private applyStylesRecursivelly;
    private addStyle;
    private selectSupportedLang;
    setTagFactory(tagFactory: TagFactory): void;
    setTagName(name: string): void;
}
