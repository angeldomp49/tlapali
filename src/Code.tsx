import { Lang, Theme } from "@makechtec/tlapali_dev";
import React from "react";
import { Translator } from "./Translator";

export const Code = (props: {theme: Theme, lang: Lang,children: any}) => {

    let translator: Translator = new Translator(props.theme, props.lang);
    let plainHTML = translator.convert(props.children);

    return(
        <div dangerouslySetInnerHTML={{__html:plainHTML}} ></div>
    );

};
