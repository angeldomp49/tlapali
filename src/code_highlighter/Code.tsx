import { Lang, Theme } from "@makechtec/tlapali_dev";
import React from "react";
import { Translator } from "./Translator";

export const Code = (props: {theme: Theme, lang: Lang,children: any}) => {

    let translator: Translator = new Translator(props.theme, props.lang);

    return(
        <>{translator.convert(props.children)}</>
    );

};
