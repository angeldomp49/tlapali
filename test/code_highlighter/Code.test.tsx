import React from 'react';
import {Code} from '../../src/code_highlighter/Code';
import renderer from 'react-test-renderer';
import { Theme, Lang } from '@makechtec/tlapali_dev';

const exampleTheme: Theme = {
    name: 'exampleTheme',
    supportedLaguages: [
        {
            name: 'exampleLang',
            styleSets: [
                {
                    rule: 'numbers',
                    styles: [
                        'color: red;'
                    ]
                }
            ]
        }
    ]
};

const exampleLang: Lang = {
    name: 'exampleLang',
    keywords: [
        'public'
    ],
    rules: [
        {
            name: 'numbers',
            pattern: '[0-9]'
        }
    ]
};

test('testCodeRender', () => {
    const tree = renderer.create(
        <Code theme={exampleTheme} lang={exampleLang} >
            3 3
        </Code>
    )
    .toJSON();
    expect(tree).toMatchSnapshot();
    // const linkElement = screen.getByText('<span style="color: red;" >3</span> <span style="color: red;" >3</span>');
    // expect(linkElement).toBeInTheDocument();
  });