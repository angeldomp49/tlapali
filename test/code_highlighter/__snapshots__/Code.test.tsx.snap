// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testCodeRender 1`] = `"<span style="color: red;" >3</span> <span style="color: red;" >3</span>"`;
